SU2_CFD/src/solvers/CSolver.cpp

```diff
@@ -2074,7 +2074,7 @@ void CSolver::AdaptCFLNumber(CGeometry **geometry,
     }
 
     su2double maxLinResid = max(linResFlow, linResTurb);
-    if (maxLinResid > 0.5) {
+    if (maxLinResid > config->GetLinear_Solver_Error()) {
       reduceCFL = true;
     }
```

Comment:
This might have to be accompanied by some changes to the best practices that @economon had established in #790. In most RANS cases that I have run, the linear solver error is almost never below the 1E-10 within the suggested 10 linear solver iterations.

Maybe the idea behind this check was to put an **upper limit** on what the linear solver error can be?

Reply:
Maybe it should be some factor larger than the linear solver tolerance then, because 0.5 just seems ridiculously high.

But I guess this raises the question of if this check is even necessary. I see the logic behind why we might want to reduce CFL if the linear solver isn't fully converging, but I feel like generally a partially converged linear solver doesn't lead to failure. Perhaps we just replace this altogether with some check that the nonlinear residual isn't increasing.

Reply:
> but I feel like generally a partially converged linear solver doesn't lead to failure

I've experienced otherwise, BCGSTAB is prone to blowing up and FGMRES may not make any progress at all if the system is too ill-conditioned.
This is not my business, but I think changes to this kind of highly empirical strategy should be accompanied by empirical evidence.

Reply:
Also in many of the examples we have the linear solver tolerance is set very low to force a fixed number of linear iterations, and so if you base the decision of reducing CFL on attaining the unobtainable...

Reply:
Something else I notice when I using CFL adaptation is that the dependence on linear convergence can cause a saw-tooth behavior, the CFL increases at max rate and then cuts down to the minimum value. I think ideally the algorithm would find the value that the linear solver can handle and hold it.

Reply:
Yeah I can see why Tom included that as a safeguard, but I don't recall seeing a dependence on linear convergence in any of the other papers I referenced when I opened this PR (e.g. https://arc.aiaa.org/doi/10.2514/6.2011-3696). Have you tested without it to see if the saw-tooth goes away? Maybe the other checks on UR and nonlinear residual are enough.
> I think ideally the algorithm would find the value that the linear solver can handle and hold it.
That's a good idea. I like that more than 0.5 or a user-specified value, which are both pretty arbitrary.